### PR TITLE
Generalized installer

### PR DESCRIPTION
**Pull request by fevo1971**

### CODE REVIEW

**Comment by fevo1971** [comment]
with this release of *get-parity.sh* distribution-specific packages are are no longer supported, users (if not uninstalling the parity package manually) might end up with inconsistent / broken packages.

File: get-parity.sh [hunk 1]
```diff
@@ -39,103 +37,56 @@ get_package() {
...
+	if [ "$PKG" = "linux" ] ; then
+	  sudo cp $TMPDIR/parity /usr/bin && sudo chmod +x /usr/bin/parity
 	fi
 
 	if [ "$PKG" = "darwin" ] ; then
-		NAME=$(basename $DOWNLOAD_FILE)
-		sudo /usr/sbin/installer -pkg $TMPDIR/$NAME -target / && \
-		sudo ln -sF /Applications/Parity\ Ethereum.app/Contents/MacOS/parity /usr/local/bin/parity
+	  sudo cp $TMPDIR/parity /usr/local/bin && sudo chmod +x /usr/local/bin/parity
 	fi
```
Comment (gabreal) [inline get-parity.sh]: if PKG is neither of both it won't be installed without notification (e.g. capital letters)

File: get-parity.sh [hunk 2]
```diff
@@ -1,30 +1,28 @@
...
 		echo "Running on Apple"
 	else
 		echo "Unknown operating system"
 		echo "Please select your operating system"
-		echo "Choices: debian - Ubuntu / Debian"
-		echo "	     linux - Other linux distro"
+		echo "Choices:"
+		echo "	     linux - any linux distro"
 		echo "	     darwin - MacOS"
 		read PKG
```
Comment (gabreal) [inline get-parity.sh]: input not validated (see below)

File: get-parity.sh [hunk 3]
```diff
@@ -39,103 +37,56 @@ get_package() {
...
-	if [ "$PKG" = "linux" -a $LIBSSL = "undef" ] ; then
-	    sudo cp $TMPDIR/parity /usr/bin
-		sudo chmod +x /usr/bin/parity
-	fi
-
-        if [ "$PKG" = "centos" ] ; then
-	    NAME=$(basename $DOWNLOAD_FILE)
-	    sudo rpm -U $TMPDIR/$NAME
+	if [ "$PKG" = "linux" ] ; then
+	  sudo cp $TMPDIR/parity /usr/bin && sudo chmod +x /usr/bin/parity
```
Comment (gabreal) [inline get-parity.sh]: why not use /usr/local/bin ?

File: get-parity.sh [hunk 4]
```diff
@@ -144,19 +95,57 @@ version_gt() {
...
+    exit 1
+  fi
+
+  # $SHA256_CHECK $TMPDIR/$DOWNLOAD_FILE 
+  IS_CHECKSUM=$($SHA256_CHECK $TMPDIR/parity | awk '{print $1}')
+  MUST_CHECKSUM=$(curl -sS $LOOKUP_URL | grep ' \[parity\]' | awk '{print $NF'})
+  # debug # echo -e "is checksum:\t $IS_CHECKSUM"
+  # debug # echo -e "must checksum:\t $MUST_CHECKSUM"
+  if [[ $IS_CHECKSUM != $MUST_CHECKSUM ]]; then
+    echo "SHA256 Checksum missmatch, abording installation"
```
Comment (gabreal) [inline get-parity.sh]: aborting

**Pull request merged**